src/common/util/util.ts

```diff
@@ -240,12 +240,22 @@ export const kTypedArrayBufferViewConstructors = Object.values(kTypedArrayBuffer
 function subarrayAsU8(
   buf: ArrayBuffer | TypedArrayBufferView,
   { start = 0, length }: { start?: number; length?: number }
-): Uint8Array {
+): Uint8Array | Uint8ClampedArray {
   if (buf instanceof ArrayBuffer) {
     return new Uint8Array(buf, start, length);
+  } else if (buf instanceof Uint8Array || buf instanceof Uint8ClampedArray) {
+    // Don't wrap in new views if we don't need to.
+    if (start === 0 && (length === undefined || length === buf.byteLength)) {
+      return buf;
+    }
+    return buf.subarray(start, length !== undefined ? start + length : undefined);
```

Comment:
nit: is there value in using `.subarray` vs `new Uint8Array` here?
IIUC, they will both give you a new view, without a new buffer allocation.
So, `new Uint8Array` might be *slightly* more concise since we can put the length arg in as-is without the ternary

Reply:
IIUC `new Uint8Array` can be used to make a view onto an ArrayBuffer, but not onto another `Uint8Array` without some complexity.

Reply:
If we use `new Uint8Array()` and don't want it to do a copy, we have to pass in the source array's buffer. The buffer may have a longer length than the `Uint8Array` view, so we'd have to guard against an undefined length extending past the bounds of the original view.

Turns out this is what the `else` statement does in this case, so I just removed the `.subarray()` path here and had it fall though to that common case.